src/lib/serviceRoutines/postUpdateContext.cpp

```diff
@@ -52,6 +53,321 @@ static char* xmlPayloadClean(const char*  payload, const char* payloadWord)
+
+  LM_M(("KZ: Forwarding request: %s", cleanPayload));
+  out = sendHttpSocket(ip,
+                       port,
+                       protocol,
+                       verb,
+                       tenant,
+                       servicePath,
+                       ciP->httpHeaders.xauthToken,
+                       resource,
+                       "application/xml",
+                       cleanPayload,
+                       false,
+                       true);
+
+  if ((out == "error") || (out == ""))
+  {
+    upcrsP->errorCode.fill(SccContextElementNotFound, "");
+    LM_E(("Runtime Error (error forwarding 'Update' to providing application)"));
+    return;
+  }
+
+
+  //
+  // 4. Parse the XML response and fill in a binary UpdateContextResponse
+  //
+  std::string  s;
+  std::string  errorMsg;
+
+  cleanPayload = xmlPayloadClean(out.c_str(), "<updateContextResponse>");
+
+  if ((cleanPayload == NULL) || (cleanPayload[0] == 0))
+  {
+    //
+    // This is really an internal error in the Context Provider
+    // It is not in the orion broker though, so 404 is returned
+    //
+    LM_W(("Other Error (context provider response to UpdateContext is empty)"));
+    upcrsP->errorCode.fill(SccContextElementNotFound, "invalid context provider response");
+    return;
+  }
+
+  //
+  // NOTE
+  // When coming from a convenience operation, such as GET /v1/contextEntities/EID/attributes/attrName,
+  // the verb/method in ciP is GET. However, the parsing function expects a POST, as if it came from a 
+  // POST /v1/updateContext. 
+  // So, here we change the verb/method for POST.
+  //
+  ParseData parseData;
+
+  ciP->verb   = POST;
+  ciP->method = "POST";
+
+  s = xmlTreat(cleanPayload, ciP, &parseData, RtUpdateContextResponse, "updateContextResponse", NULL, &errorMsg);
+  if (s != "OK")
+  {
+    LM_W(("Internal Error (error parsing reply from prov app: %s)", errorMsg.c_str()));
+    upcrsP->errorCode.fill(SccContextElementNotFound, "");
+    parseData.upcr.res.release();
+    parseData.upcrs.res.release();
+    return;
+  }
+
+
+  //
+  // 5. Render UpdateContextResponse from the providing application
+  //
+  char portV[16];
+  snprintf(portV, sizeof(portV), "%d", port);
+
+  // Fill in the response from the redirection into the response
+  upcrsP->fill(&parseData.upcrs.res);
+
+
+  //
+  // 6. 'Fixing' StatusCode
+  //
+  if (upcrsP->errorCode.code == SccNone)
+  {
+    upcrsP->errorCode.fill(SccOk);
+  }
+
+  if ((upcrsP->contextElementResponseVector.size() == 1) && (upcrsP->contextElementResponseVector[0]->statusCode.code == SccContextElementNotFound))
+  {
+    upcrsP->errorCode.fill(SccContextElementNotFound);
+  }
+
+  
+  //
+  // 7. Freeing memory
+  //
+  parseData.upcr.res.release();
+  parseData.upcrs.res.release();
+}
+
+
+
+/* ****************************************************************************
+*
+* foundAndNotFound - 
```

Comment:
Yeah, perhaps.
To me it is very clear right now, but in a few weeks ...
foundAndNotFoundAttributeSeparation?

It is all about adding a ContextElementResponse for attributes that are Not Found - they need a different statusCode. 

There are three different cases:
1. All attributes are Found - nothing to touch
2. All attributes are Not Found - the statusCode is changed from 200 to 404
3. A mix of attributes Found and Not Found - here I need a new ContextElementResponse and all the Not Found attributes must be moved to the new ContextElementResponse (that has statusCode 404).

If you find a good name ...